### PR TITLE
Slimmified CI 

### PR DESCRIPTION
**Pull request by umgefahren**
# Description

I removed some steps from the CI to improve the peformance. This is possible, because GitHub Actions runners now come with Rust support.

## Links to any relevant issues

Origin: #2900 

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -127,19 +111,10 @@ jobs:
...
-        toolchain: stable
-        override: true
-        components: clippy
-
     - uses: Swatinem/rust-cache@6720f05bc48b77f96918929a9019fb2203ff71f8 # v2.0.0
 
-    - name: Run cargo clippy
-      uses: actions-rs/cargo@844f36862e911db73fe0815f00a4a2602c279505 # v1.0.3
-      with:
-        command: custom-clippy # cargo alias to allow reuse of config locally
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:142]: We are running an alias here, please retain that (with the comment!).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -156,12 +131,6 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-
     - uses: Swatinem/rust-cache@6720f05bc48b77f96918929a9019fb2203ff71f8 # v2.0.0
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml]: Can we bump to this v2 as well then?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -127,19 +111,10 @@ jobs:
...
     - uses: actions/checkout@v3
 
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-        components: clippy
-
     - uses: Swatinem/rust-cache@6720f05bc48b77f96918929a9019fb2203ff71f8 # v2.0.0
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml]: Here too!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -102,12 +92,6 @@ jobs:
 
     - uses: actions/checkout@v3
 
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-
     - uses: Swatinem/rust-cache@6720f05bc48b77f96918929a9019fb2203ff71f8 # v2.0.0
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml]: And here.

**Review by thomaseizinger — COMMENTED** [review]
Thanks! I am in favor of this.

I don't know why we are referencing actions by hash instead of version. @mxinden Do you know?

**Comment by mxinden** [comment]
> I don't know why we are referencing actions by hash instead of version. @mxinden Do you know?

Reasoning is here:

> I pinned the third-party action versions to the exact commit SHAs. We follow the same strategy in protocol go repositories to ensure the code we use doesn't change.

https://github.com/libp2p/rust-libp2p/pull/2424

I don't have a strong opinion here. Maybe @galargh does?

**Comment by galargh** [comment]
Yes, exactly! Pinning actions by hash is a mitigation for a potential supply chain attack. We try to do that for all actions that are neither from PL orgs nor https://github.com/actions.

**Comment by thomaseizinger** [comment]
Okay, thanks for clarifying! Let's go back to commit hashes then please @umgefahren!

Plus, https://github.com/libp2p/rust-libp2p/pull/2917#discussion_r974884841 needs to be addressed before we can merge this!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (umgefahren) [reply .github/workflows/ci.yml:142]: But why are you running an alias here? I don't quite understand the assignment.

Comment (thomaseizinger) [reply .github/workflows/ci.yml:142]: The alias is a workaround for clippy still not having a config file.

It is defined here: https://github.com/libp2p/rust-libp2p/blob/master/.cargo/config.toml#L3

This allows us to allow certain lints across the codebase without having to duplicate this in a bunch of places :)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -20,16 +20,16 @@ jobs:
...
         access_token: ${{ github.token }}
 
     - name: Install Protoc
-      uses: arduino/setup-protoc@v1
+      uses: arduino/setup-protoc@31a047f6ff1b725b36c561770f81012e2f165926
 
     - uses: actions/checkout@v3
 
-    - uses: Swatinem/rust-cache@6720f05bc48b77f96918929a9019fb2203ff71f8 # v2.0.0
+    - uses: Swatinem/rust-cache@76686c56f2b581d1bb5bda44b51f7e24bd9b8b8e
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:32]: As far as I can see, this hash is not yet part of a released version. I'd prefer if we just not touch this line at all and revert back to exactly what we had :)

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -20,16 +20,16 @@ jobs:
     steps:
 
     - name: Cancel Previous Runs
-      uses: styfle/cancel-workflow-action@bb6001c4ea612bf59c3abfc4756fbceee4f870c7 # 0.10.0
+      uses: styfle/cancel-workflow-action@0.10.0 # 0.10.0
       with:
         access_token: ${{ github.token }}
 
     - name: Install Protoc
-      uses: arduino/setup-protoc@v1
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:28]: It is my understanding that [`v1`](https://github.com/arduino/setup-protoc/tree/v1) would actually correspond to https://github.com/arduino/setup-protoc/commit/64c0c85d18e984422218383b81c52f8b077404d3.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -20,16 +20,16 @@ jobs:
     steps:
 
     - name: Cancel Previous Runs
-      uses: styfle/cancel-workflow-action@bb6001c4ea612bf59c3abfc4756fbceee4f870c7 # 0.10.0
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:23]: Let's revert this too please.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thank you!

I think the original idea of the PR - to get rid of `actions-rs/toolchain` is good. To stay focused on that, can we revert the other changes to what they were? I think the old commit hashes been set exactly to the hash of the releases before and now we are pointing to some commit on master.

**Comment by umgefahren** [comment]
I will take another go at this later. Closing for now.

**Pull request closed without merging**